File: AILang/benchmarks/c/gauntlet_numeric.c
#include <stdint.h>
#include <stdio.h>

static int64_t numeric_mix(int64_t seed, int64_t iterations) {
    int64_t x = seed % 1000003LL;
    int64_t y = 911382323LL % 1000003LL;
    int64_t z = 972663749LL % 1000003LL;
    int64_t acc = 0;
    for (int64_t i = 0; i < iterations; ++i) {
        x = (x * 110351LL + 12345LL + i) % 1000003LL;
        y = (y + x * 31LL + i * 17LL) % 1000033LL;
        if (x > y) {
            z = (z + x - y + 97LL) % 1000037LL;
        } else {
            z = (z + y - x + 193LL) % 1000037LL;
        }
        if (z % 7LL == 0LL) {
            acc = (acc + z * 3LL + x) % 1000000007LL;
        } else {
            acc = (acc + y * 5LL + z) % 1000000007LL;
        }
    }
    return acc;
}

int main(void) {
    printf("%lld\n", (long long)numeric_mix(1234567LL, 8000000LL));
    return 0;
}

File: AILang/benchmarks/rust/gauntlet_numeric.rs
fn numeric_mix(seed: i64, iterations: i64) -> i64 {
    let mut x = seed % 1_000_003;
    let mut y = 911_382_323_i64 % 1_000_003;
    let mut z = 972_663_749_i64 % 1_000_003;
    let mut acc = 0_i64;
    let mut i = 0_i64;
    while i < iterations {
        x = (x * 110_351 + 12_345 + i) % 1_000_003;
        y = (y + x * 31 + i * 17) % 1_000_033;
        if x > y {
            z = (z + x - y + 97) % 1_000_037;
        } else {
            z = (z + y - x + 193) % 1_000_037;
        }
        if z % 7 == 0 {
            acc = (acc + z * 3 + x) % 1_000_000_007;
        } else {
            acc = (acc + y * 5 + z) % 1_000_000_007;
        }
        i += 1;
    }
    acc
}

fn main() {
    println!("{}", numeric_mix(1_234_567, 8_000_000));
}
